.travis.yml

```diff
@@ -16,19 +16,14 @@ after_success:
 before_deploy:
   - docker push chengjoanna1729/todo-app:latest
   - docker pull chengjoanna1729/todo-app

```

Comment:
This is a bit odd. You pull after you push? You should pull before building (to take advantage of caching).

Reply:
what do you mean? as in changing the after-success section to:
```
after_success:
  - echo $DOCKER_PASSWORD | docker login --username chengjoanna1729 --password-stdin
  - docker pull chengjoanna1729/todo-app
  - docker build --target production --tag chengjoanna1729/todo-app:latest .
  - docker tag chengjoanna1729/todo-app:latest chengjoanna1729/todo-app:$TRAVIS_COMMIT
  - docker push chengjoanna1729/todo-app:$TRAVIS_COMMIT
before_deploy:
  - docker push chengjoanna1729/todo-app:latest
 ```
? will the build step then automatically be able to use the cache?